src/features/Folding.ts

```diff
@@ -0,0 +1,415 @@
+    }
+
+    // Finding blocks of comment tokens is more complicated as the newline characters are not
+    // classed as comments.  To workaround this we search for the comment character `#` scope name
+    // "punctuation.definition.comment.powershell" and then determine contiguous line numbers from there
+    private matchBlockCommentScopeElements(tokens,
+                                           document: vscode.TextDocument) {
+        const result = [];
+
+        const emptyLine = new RegExp("^[\\s]+$");
+
+        let startLine: number = -1;
+        let nextLine: number = -1;
+
+        tokens.forEach((token, index) => {
+            if (token.scopes.includes("punctuation.definition.comment.powershell")) {
+                // The punctuation.definition.comment.powershell token matches new-line comments
+                // and inline comments e.g. `$x = 'foo' # inline comment`.  We are only interested
+                // in comments which begin the line i.e. no preceeding text
+                if (emptyLine.test(this.preceedingText(token.startIndex, document))) {
+                    const lineNum = this.lineAtOffest(token.startIndex, document);
+                    // A simple pattern for keeping track of contiguous numbers in a known
+                    // sorted array
+                    if (startLine === -1) {
+                        startLine = lineNum;
+                        nextLine = lineNum + 1;
+                    } else {
+                        if (lineNum === nextLine) {
+                            nextLine = lineNum + 1;
+                        } else {
+                            result.push(new MatchedToken(null,
+                                                         null,
+                                                         startLine,
+                                                         nextLine - 1,
+                                                         MatchType.Comment,
+                                                         document));
+                            startLine = lineNum;
+                            nextLine = lineNum + 1;
+                        }
+                    }
+                }
+            }
+        });
+
+        // If we exist the token array and we're still processing comment lines, then the
+        // comment block simply ends at the end of document
+        if (startLine !== -1) {
+            result.push(new MatchedToken(null, null, startLine, nextLine - 1, MatchType.Comment, document));
+        }
+
+        return result;
+    }
+
+    // Given a zero based offset, find the line text after it in the document
+    private subsequentText(offset: number,
+                           document: vscode.TextDocument): string {
+        const startPos = document.positionAt(offset);
+        // We don't know how long the line is so just return a really long one.
+        const endPos = startPos.translate(0, 1000);
```

Comment:
you can get the whole line with:

```typescript
document.lineAt(document.positionAt(offset)).rangeIncludingLineBreak
//or
document.lineAt(document.positionAt(offset)).range // for no LineBreak
```

from that can we accurately get the start and end position?

Reply:
Fixed